issues:add in list #3009,#3010, correction blank in #3011

harrykar opened the pull request:
## What does this PR do?
Add Resource(s) | Improve Repo

## For resources
### Description 
added:
Object-oriented Programming in JavaTM Textbook by Rick Halterman #3009
Introduction to Computer Science by Robert Sedgewick and Kevin Wayne. #3010
corrected:
link not displayed correctly(displayed with makdown)
deleted blanc ...Problem Solving] (https://archive.... in Java, Java, Java Object-Oriented Problem Solving by R. Morelli and R.Walde #3011

### Why is this valuable (or not)
Knowledge is ever valuable
### How do we know it's really free?
Source are accademic Websites
### For book lists, is it a book?

### Checklist:
- [x ] Not a duplicate
- [ x] Included author(s) if appropriate
- [x ] Lists are in alphabetical order
- [ x] Needed indications added (PDF, access notes, under construction)

harrykar commented:
I did all correct dunno why that conflict. Someone from staff need here i guess

eshellman commented:
To avoid merge conflicts, always pull changes from the source before committing.

eshellman commented:
Also, the "staff" is you, and a thousand others!

harrykar commented:
> To avoid merge conflicts, always pull changes from the source before committing.

I guess more users are afraid of git cause it's initial high step learn curve
Ok let's do a bit of git school 
1. as you have mentioned first of all we need do a pull from the source centralized repo to get all changes others did meanwhile we was not connected to the central repo but for me the problem was  
1.a. i had forked the central/source repo and i have (more than one time) requested here how i can update the fork(after i did the first PR) but i had get no answer.  I had to search some resources(Note: was useful to update Howto contribute with a more concrete tutorial) to find that synthesized(by me) solution

a. Add the central/source repo to our local repo(cloned from the fork ) 
$ **git remote add upstream** git@github.com:**EbookFoundation/free-programming-books.git**

Now we've our forked(origin) , local,  and central(upstream) repo 
$ git remote -v
origin	git@github.com:harrykar/free-programming-books.git (fetch)
origin	git@github.com:harrykar/free-programming-books.git (push)
upstream	git@github.com:EbookFoundation/free-programming-books.git (fetch)
upstream	git@github.com:EbookFoundation/free-programming-books.git (push)


b. We have to move to our local master branch(in case we're in another branch)
$ **git checkout master**
Switched to branch 'master'


c. fetch updates from the central/source(upstream) repo into our local repo
$ **git fetch upstream**
remote: Enumerating objects: 40, done.
remote: Counting objects: 100% (40/40), done.
remote: Compressing objects: 100% (16/16), done.
remote: Total 44 (delta 27), reused 35 (delta 24), pack-reused 4
Unpacking objects: 100% (44/44), done.
From github.com:EbookFoundation/free-programming-books
 * [new branch]      close-2491 -> upstream/close-2491
 * [new branch]      deprecate-javascript-pages -> upstream/deprecate-javascript-pages
 * [new branch]      eshellman-patch-1 -> upstream/eshellman-patch-1
 * [new branch]      how-to     -> upstream/how-to
 * [new branch]      master     -> upstream/master
 * [new branch]      open-source-ecosystem -> upstream/open-source-ecosystem
 * [new branch]      reformat-screan-pod-casts -> upstream/reformat-screan-pod-casts
 * [new branch]      remove-frameworks-pt-BR -> upstream/remove-frameworks-pt-BR
 * [new branch]      requested-emails -> upstream/requested-emails
 * [new branch]      revert-3021-master -> upstream/revert-3021-master
 * [new branch]      standardize-wikibooks -> upstream/standardize-wikibooks

Let's do a glance in our branches in all our (forked(origin) , local,  and central(upstream) ) repos 
$ **git branch -va**
  harrykar                                    9ca0f12 issues:add in list #3009,#3010, correction blank in #3011
* master                                      9ca0f12 issues:add in list #3009,#3010, correction blank in #3011
  remotes/origin/HEAD                         -> origin/master
  remotes/origin/close-2491                   5a1a911 add princeton algorithms
  remotes/origin/deprecate-javascript-pages   6a461e9 deprecate the javascript frameworks pages
  remotes/origin/eshellman-patch-1            20b37e4 remove js framework link
  remotes/origin/how-to                       8e35929 add resource titles and delint
  remotes/origin/master                       9ca0f12 issues:add in list #3009,#3010, correction blank in #3011
  remotes/origin/open-source-ecosystem        66ad3ec Open source ecosystem
  remotes/origin/reformat-screan-pod-casts    52c0239 trial and error
  remotes/origin/remove-frameworks-pt-BR      0b66ec3 move js resources into books or casts as appropriate
  remotes/origin/requested-emails             6f25b36 remove dead link
  remotes/origin/standardize-wikibooks        5047c7f Merge branch 'master' into standardize-wikibooks
  remotes/upstream/close-2491                 5a1a911 add princeton algorithms
  remotes/upstream/deprecate-javascript-pages 6a461e9 deprecate the javascript frameworks pages
  remotes/upstream/eshellman-patch-1          20b37e4 remove js framework link
  remotes/upstream/how-to                     8e35929 add resource titles and delint
  remotes/upstream/master                     f50175e Revert 3021 master (#3025)
  remotes/upstream/open-source-ecosystem      66ad3ec Open source ecosystem
  remotes/upstream/reformat-screan-pod-casts  52c0239 trial and error
  remotes/upstream/remove-frameworks-pt-BR    0b66ec3 move js resources into books or casts as appropriate
  remotes/upstream/requested-emails           6f25b36 remove dead link
  remotes/upstream/revert-3021-master         eaadb3a remove space, readd javajavajava
  remotes/upstream/standardize-wikibooks      5047c7f Merge branch 'master' into standardize-wikibooks

d. time to incorporate the central/source repo's changes to our local repo(make sure you're in master branch) 
Here i notice ine can use merge or rebase(actually dunno what is better if someone know please explain what of the two is better and why). I use rebase that time 
$ **git rebase upstream/master**
First, rewinding head to replay your work on top of it...
Applying: issues:add in list #3009,#3010, correction blank in #3011
Using index info to reconstruct a base tree...
Falling back to patching base and 3-way merge...
Auto-merging free-programming-books.md
  
e. final step is update our fork from our local repo
$ **git push origin master**
To git@github.com:harrykar/free-programming-books.git
 ! [rejected]        master -> master (non-fast-forward)
error: failed to push some refs to 'git@github.com:harrykar/free-programming-books.git'
To prevent you from losing history, non-fast-forward updates were rejected
Merge the remote changes before pushing again.  See the 'Note about
fast-forwards' section of 'git push --help' for details.

not sure i got the semantics of the last's output step and precisely  what the quoted part mean The update to remote is failed? who read here Explain further please 

>  ! [rejected]        master -> master (non-fast-forward)
> error: failed to push some refs to 'git@github.com:harrykar/free-programming-books.git'
> To prevent you from losing history, non-fast-forward updates were rejected
> Merge the remote changes before pushing again.  See the 'Note about
> fast-forwards' section of 'git push --help' for details.

harrykar commented:
> Also, the "staff" is you, and a thousand others! (In this case, the problem was I messed up the revert.)

Yeah i know so let's say "staff" is who is more experienced and can help others :)

eshellman commented:
I use the github desktop app and resort to the command line only when doing brain surgery. you just need to alphabetize the list

harrykar commented:
> not sure i got the semantics of the last's output step and precisely what the quoted part mean The update to remote is failed? who read here Explain further please, I ncase the update succeed what mean "Merge the remote changes before pushing again."?
>     $git push origin master
>     To git@github.com:harrykar/free-programming-books.git
>     ! [rejected] master -> master (non-fast-forward)
>     error: failed to push some refs to 'git@github.com:harrykar/free-programming-books.git'
>     To prevent you from losing history, non-fast-forward updates were rejected
>     Merge the remote changes before pushing again. See the 'Note about
>     fast-forwards' section of 'git push --help' for details.

That message mean the push is failed because the remote(origin/master) has changes respect of our local master as is shown 
```$ git status 
# On branch master
# Your branch and 'origin/master' have diverged,
# and have 13 and 2 different commit(s) each, respectively.
#
nothing to commit (working directory clean)
$ git status 
# On branch master
# Your branch and 'origin/master' have diverged,
# and have 13 and 2 different commit(s) each, respectively.
#
nothing to commit (working directory clean)
```
To remedy this situation, we need to synchronize with our remote(origin/master)  repository. Then, we’ll be able to incorporate origins changes to our master branch(through rebase) and finally push our changes in the usual fashion.
```
git fetch origin master
remote: Enumerating objects: 2, done.
remote: Counting objects: 100% (2/2), done.
remote: Total 3 (delta 2), reused 2 (delta 2), pack-reused 1
Unpacking objects: 100% (3/3), done.
From github.com:harrykar/free-programming-books
 * branch            master     -> FETCH_HEAD
```
```
$ git rebase origin/master
First, rewinding head to replay your work on top of it...
Applying:  [The Arduino Platform and C Programming](https://www.coursera.org/learn/arduino-platform) (#3012)
Applying: Enable emoji on GitHub Pages (#3013)
Applying: Remove "under construction" (#3014)
Applying: Fixed old dead link in german (#3015)
Applying: remove dead links (#3018)
Applying: remove dead link (#3019)
Applying: remove-dead-italian-links (#3017)
Applying: Added resources (#3022)
Applying: Added Git and Github course (#3023)
Applying: remove dead link from free-programming-books-tr.md (#3024)
Applying: Revert 3021 master (#3025)
Using index info to reconstruct a base tree...
<stdin>:23: trailing whitespace.
* [The Road to learn React - Build a Hacker News App along the Way](https://leanpub.com/the-road-to-learn-react) - Robin Wieruch *(Leanpub account or valid email requested)* [markdown](https://github.com/the-road-to-learn-react/the-road-to-learn-react/tree/master/manuscript)  
<stdin>:32: trailing whitespace.
    * [The Symfony Getting started 3.4.x](https://symfony.com/pdf/Symfony_getting_started_3.4.pdf) (PDF)    
warning: 2 lines add whitespace errors.
Falling back to patching base and 3-way merge...
Auto-merging free-programming-books.md
Applying: issues:add in list #3009,#3010, correction blank in #3011
Using index info to reconstruct a base tree...
Falling back to patching base and 3-way merge...
No changes -- Patch already applied.
```
```
$ git push origin master
To git@github.com:harrykar/free-programming-books.git
 ! [rejected]        master -> master (non-fast-forward)
error: failed to push some refs to 'git@github.com:harrykar/free-programming-books.git'
To prevent you from losing history, non-fast-forward updates were rejected
Merge the remote changes before pushing again.  See the 'Note about
fast-forwards' section of 'git push --help' for details.
```
yet not in sync ; lets use the same procedure another one time 

```
$ git fetch 
From github.com:harrykar/free-programming-books
   9ca0f12..268e94d  master     -> origin/master
```

```
$ git rebase origin
First, rewinding head to replay your work on top of it...
Applying: Revert 3021 master (#3025)
```

```
$ git push
Counting objects: 5, done.
Delta compression using up to 2 threads.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (3/3), 645 bytes, done.
Total 3 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To git@github.com:harrykar/free-programming-books.git
   268e94d..46876e2  master -> master
```
Finally now after have updated our remote origin (fork) with the latest changes of the remote upstream (the source/central repository) trough our local cloned(from the remote origin) repository we can continue to collaborate further

harrykar commented:
> I use the github desktop app and resort to the command line only when doing brain surgery. you just need to alphabetize the list

Seems that git is much more than a DCVS reminds me something about abstraction, data types etc i mean seems ,ore a huge conceptual tool than a simple CVS 
Now is late i have to go further trough the CLI after all git is a CLI tool. GUI can wait :D

harrykar commented:
> I use the github desktop app and resort to the command line only when doing brain surgery. you just need to alphabetize the list

First all first in https://desktop.github.com/ i can't find a package for linux if i don't miss something;On one hand that's not good(divide trough users of different OS's?!) but as usual  i have to go further trough the CLI after all git is a CLI tool. Actually i feel  comfortable with my [geany editor](https://en.wikipedia.org/wiki/Geany) + his incorporated shell plugin  

Pull Requests can be made also trough CLI by  [hub](https://github.com/github/hub) (a github's git wrapper tool) that have to be downloaded and "installed" locally but for now i use the browser for that

eshellman commented:
There's no wiki on this repo, but feel free to either

1. write a tutorial on your own wiki and link to is from CONTRIBUTING.md
2. write a tutorial page and propose adding it to the repo in a PR. 

I expect you'll want help with editing - request help in an issue.

harrykar commented:
> There's no wiki on this repo, but feel free to either

That's not a wiki?
https://github.com/EbookFoundation/free-programming-books/wiki

eshellman commented:
So there is! See my modified comment - I'll create a wiki team and add you if needed.

harrykar commented:
Thanks @eshellman 
I guess the most appropriate place is the Wiki (if am not wrong that's tracked too )
I already posted the first draft (i guess is almost ready as it is except of language errors -I'm not a native English-) take a look at and feel free to change

eshellman commented on free-programming-books.md:
```diff
@@ -1387,6 +1394,7 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Core Servlets and JavaServer Pages, 2nd Ed. (2003)](http://pdf.coreservlets.com) - Marty Hall and Larry Brown
 * [Data Structures in Java for the Principled Programmer (2007)](http://dept.cs.williams.edu/~bailey/JavaStructures/Book_files/JavaStructures.pdf) - Duane A. Bailey (PDF)
 * [Google's Java Style Guide](https://google.github.io/styleguide/javaguide.html)
+* [Introduction to Computer Science](https://introcs.cs.princeton.edu/java/cs/) - by Robert Sedgewick and Kevin Wayne (HTML)
```
This "booksite" is not the book. We can include it if you think it's still useful, but should add a notation.

harrykar replied:
Nowadays Books exists in some formats. This web page IS the book (in HTML form hence the (HTML) instead of (PDF) or (OTHER FORMAT))

harrykar replied:
By the way i have a question
Suppose there is an author's  page in which  other than author's textbook refer other useful informations like exercises other subject matter related links bibliography etc 
In that case IMHO is more useful/valuable refer in the books list the author's page(that contains the author's PDF) than the author's textbook link
Such typical use cases are Uni's profs pages

eshellman replied:
We have lots of HTML websites-as-books. This isn't one of those. Its a free-website-to-go-with-book-that-you've-bought. Nothing wrong with that, but it's not the not-free-book-that-you-buy.

eshellman replied:
We often link to a book webpage where links to the book file are presented along with related material.

harrykar replied:
> We have lots of HTML websites-as-books. This isn't one of those. Its a free-website-to-go-with-book-that-you've-bought. Nothing wrong with that, but it's not the not-free-book-that-you-buy.

Right now i am in house and have the book at hand. I was pretty sure about that. The prof page is exactly the book more in their page is much more better organized than in printed form

harrykar replied:
> We often link to a book webpage where links to the book file are presented along with related material.

That's the right thing i guess :+1:

eshellman commented on free-programming-books.md:
```diff
@@ -1402,6 +1410,8 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Neural Network Development with Java](https://www.packtpub.com/packt/free-ebook/neural-networks-java) - Alan M. F. Souza and Fabio M. Soares (email address *requested*, not required)
 * [Object-Oriented vs. Functional Programming](http://www.oreilly.com/programming/free/object-oriented-vs-functional-programming.csp) - Richard Warburton (email address *requested*, not required)
+* [Object-oriented Programming in JavaTM Textbook](http://computing.southern.edu/halterman/OOPJ/) - Rick Halterman (PDF per Chapter)
+* [Object Oriented Programming using Java](https://bookboon.com/en/object-oriented-programming-using-java-ebook) - Simon Kendal (PDF)
```
Look at other bookboon links to see the notation we require.

eshellman commented on free-programming-books.md:
```diff
@@ -35,6 +35,7 @@
 * [Delphi / Pascal](#delphi--pascal)
 * [DTrace](#dtrace)
 * [Elasticsearch](#elasticsearch)
+* [Eiffel](#eiffel)
```
Alphabetization: "Ei" < "El"

harrykar replied:
Damn it. Tell me you got it intuitively or was noted somewhere(as do a compiler)?

eshellman commented on free-programming-books.md:
```diff
@@ -1402,6 +1410,8 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Neural Network Development with Java](https://www.packtpub.com/packt/free-ebook/neural-networks-java) - Alan M. F. Souza and Fabio M. Soares (email address *requested*, not required)
 * [Object-Oriented vs. Functional Programming](http://www.oreilly.com/programming/free/object-oriented-vs-functional-programming.csp) - Richard Warburton (email address *requested*, not required)
+* [Object-oriented Programming in JavaTM Textbook](http://computing.southern.edu/halterman/OOPJ/) - Rick Halterman (PDF per Chapter)
+* [Object Oriented Programming using Java](https://bookboon.com/en/object-oriented-programming-using-java-ebook) - Simon Kendal (PDF)
```
Alphabetization: "Object " < "Object-"

harrykar replied:
LOL seems tough. hope there's an automa that can give us all that error indications

eshellman replied:
There is. It's Travis. Click on "details"

> continuous-integration/travis-ci/pr — The Travis CI build failed              Details

harrykar replied:
> There is. It's Travis. Click on "details"
> 
> > continuous-integration/travis-ci/pr — The Travis CI build failed              Details
I can't see a details  button where is it? 
in that page somewhere between our conversations i see
```
harrykar and others added some commits 8 days ago
harrykar
issue #3011 Java, Java, Java Object-Oriented Problem Solving by R. Mo…
e10717d
@eshellman
harrykar
Revert 3021 master (#3025)
f7d7261
harrykar
issues:add in list #3009,#3010, correction blank in #3011
227a313
harrykar
issue #3011 Java, Java, Java Object-Oriented Problem Solving by R. Mo…
45ad31c
harrykar
add 2 books, create [Eiffel](#eiffel),
898d962
harrykar
correct merge annotations floated around
166fb2a
harrykar
correct add one blankline before ### Subversion
1d291eb

```
the last part is the SHA1 and is a link you mean that one?

harrykar replied:
Don't mind I found it :)

eshellman commented on free-programming-books.md:
```diff
@@ -2435,6 +2445,7 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Programming in Standard ML '97](http://homepages.inf.ed.ac.uk/stg/NOTES/) - Stephen Gilmore, University of Edinburgh
 * [Programming in Standard ML, DRAFT](http://www.cs.cmu.edu/~rwh/isml/book.pdf) - Robert Harper (PDF)
 * [SML# Document](http://www.pllab.riec.tohoku.ac.jp/smlsharp/docs/3.0/en/manual.xhtml) - Atsushi Ohori, Katsuhiro Ueno
+* [The Definition of Standard ML (Revised)](http://sml-family.org/sml97-defn.pdf) - SMLFamily GitHub project
```
Add pdf notation

harrykar replied:
ok

harrykar commented:
Uffa!! After dunno how many 6-7 continous edit>add>commit>squash>push >PR's>Travis linter indications finally i was successful 
GREAT :D

eshellman commented:
Thanks for persisting!

harrykar commented:
> Thanks for persisting!

Really Thanks to you a big hug @eshellman

harrykar commented:
**Strange thing**
Today after a  brand fresh _git fetch upstream_
```
harrykar@harrysas:~/git_repos/free-programming-books$ git status 
# On branch master
nothing to commit (working directory clean)
harrykar@harrysas:~/git_repos/free-programming-books$ git fetch upstream
remote: Enumerating objects: 9, done.
remote: Counting objects: 100% (9/9), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 9 (delta 3), reused 2 (delta 2), pack-reused 0
Unpacking objects: 100% (9/9), done.
From github.com:EbookFoundation/free-programming-books
   1180056..35b7d17  master     -> upstream/master
harrykar@harrysas:~/git_repos/free-programming-books$ git rebase upstream/master
First, rewinding head to replay your work on top of it...
Applying: issue #3011 Java, Java, Java Object-Oriented Problem Solving by R. Morelli and R.Walde
Using index info to reconstruct a base tree...
Falling back to patching base and 3-way merge...
Auto-merging free-programming-books.md
Applying: issues:add in list #3009,#3010, correction blank in #3011
Using index info to reconstruct a base tree...
Falling back to patching base and 3-way merge...
Auto-merging free-programming-books.md
CONFLICT (content): Merge conflict in free-programming-books.md
Recorded preimage for 'free-programming-books.md'
Failed to merge in the changes.
Patch failed at 0002 issues:add in list #3009,#3010, correction blank in #3011

When you have resolved this problem run "git rebase --continue".
If you would prefer to skip this patch, instead run "git rebase --skip".
To restore the original branch and stop rebasing run "git rebase --abort".
```
i face a merge conflict and on investigating on it i saw exactly the same lines i deleted yesterday meanwhile i struggle to pass my revision to Travis. 
How that is possible? If i deleted that lines in the same place and commit the corrected revision to upstream(your Central repository) and you integrated my revision on your master who/what have reintroduced the bad revision again in circulation?  

Precisely between l.1982-l.1985 
<<<<<<< HEAD
=======

>>>>>>> issue #3011 Java, Java, Java Object-Oriented Problem Solving by R. Morelli and R.Walde

harrykar commented:
**tutorial?!**
```
harrykar@harrysas:~/git_repos/free-programming-books$ git status 
# On branch master
nothing to commit (working directory clean)

harrykar@harrysas:~/git_repos/free-programming-books$ git fetch upstream
remote: Enumerating objects: 9, done.
remote: Counting objects: 100% (9/9), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 9 (delta 3), reused 2 (delta 2), pack-reused 0
Unpacking objects: 100% (9/9), done.
From github.com:EbookFoundation/free-programming-books
   1180056..35b7d17  master     -> upstream/master

harrykar@harrysas:~/git_repos/free-programming-books$ git rebase upstream/master
First, rewinding head to replay your work on top of it...
Applying: issue #3011 Java, Java, Java Object-Oriented Problem Solving by R. Morelli and R.Walde
Using index info to reconstruct a base tree...
Falling back to patching base and 3-way merge...
Auto-merging free-programming-books.md
Applying: issues:add in list #3009,#3010, correction blank in #3011
Using index info to reconstruct a base tree...
Falling back to patching base and 3-way merge...
Auto-merging free-programming-books.md
CONFLICT (content): Merge conflict in free-programming-books.md
Recorded preimage for 'free-programming-books.md'
Failed to merge in the changes.
Patch failed at 0002 issues:add in list #3009,#3010, correction blank in #3011

When you have resolved this problem run "git rebase --continue".
If you would prefer to skip this patch, instead run "git rebase --skip".
To restore the original branch and stop rebasing run "git rebase --abort".

harrykar@harrysas:~/git_repos/free-programming-books$ git status 
# Not currently on any branch.
# Unmerged paths:
#   (use "git reset HEAD <file>..." to unstage)
#   (use "git add/rm <file>..." as appropriate to mark resolution)
#
#	both modified:      free-programming-books.md
#
no changes added to commit (use "git add" and/or "git commit -a")

after modified(resolved)

harrykar@harrysas:~/git_repos/free-programming-books$ git status 
# Not currently on any branch.
# Unmerged paths:
#   (use "git reset HEAD <file>..." to unstage)
#   (use "git add/rm <file>..." as appropriate to mark resolution)
#
#	both modified:      free-programming-books.md
#
no changes added to commit (use "git add" and/or "git commit -a")

Then, tell Git you’re done
resolving the conflict with the git add command:
git add <file>
That’s right; all you have to do is stage the conflicted file to mark it as resolved.

harrykar@harrysas:~/git_repos/free-programming-books$ git add .

harrykar@harrysas:~/git_repos/free-programming-books$ git status 
# Not currently on any branch.
# Changes to be committed:
#   (use "git reset HEAD <file>..." to unstage)
#
#	modified:   free-programming-books.md
#
harrykar@harrysas:~/git_repos/free-programming-books$ 

harrykar@harrysas:~/git_repos/free-programming-books$ git rebase --continue
Applying: issues:add in list #3009,#3010, correction blank in #3011
Applying: Revert 3021 master (#3025)
Applying: issue #3011 Java, Java, Java Object-Oriented Problem Solving by R. Morelli and R.Walde
Applying: add 2 books, create [Eiffel](#eiffel),
Using index info to reconstruct a base tree...
Falling back to patching base and 3-way merge...
Auto-merging free-programming-books.md
CONFLICT (content): Merge conflict in free-programming-books.md
Failed to merge in the changes.
Patch failed at 0005 add 2 books, create [Eiffel](#eiffel),

When you have resolved this problem run "git rebase --continue".
If you would prefer to skip this patch, instead run "git rebase --skip".
To restore the original branch and stop rebasing run "git rebase --abort".

harrykar@harrysas:~/git_repos/free-programming-books$ git status 
# Not currently on any branch.
# Unmerged paths:
#   (use "git reset HEAD <file>..." to unstage)
#   (use "git add/rm <file>..." as appropriate to mark resolution)
#
#	both modified:      free-programming-books.md
#
no changes added to commit (use "git add" and/or "git commit -a")

harrykar@harrysas:~/git_repos/free-programming-books$ git add .

harrykar@harrysas:~/git_repos/free-programming-books$ git rebase --continue
Applying: add 2 books, create [Eiffel](#eiffel),
Applying: swap l.1413 and l.1412%B
Using index info to reconstruct a base tree...
Falling back to patching base and 3-way merge...
Auto-merging free-programming-books.md
CONFLICT (content): Merge conflict in free-programming-books.md
Failed to merge in the changes.
Patch failed at 0006 swap l.1413 and l.1412%B

When you have resolved this problem run "git rebase --continue".
If you would prefer to skip this patch, instead run "git rebase --skip".
To restore the original branch and stop rebasing run "git rebase --abort".

harrykar@harrysas:~/git_repos/free-programming-books$ git add .

harrykar@harrysas:~/git_repos/free-programming-books$ git rebase --continue
Applying: swap l.1413 and l.1412%B
Recorded resolution for 'free-programming-books.md'.

update forked repo

harrykar@harrysas:~/git_repos/free-programming-books$ git push -f origin master
Counting objects: 259, done.
Delta compression using up to 2 threads.
Compressing objects: 100% (162/162), done.
Writing objects: 100% (235/235), 51.22 KiB, done.
Total 235 (delta 158), reused 116 (delta 72)
remote: Resolving deltas: 100% (158/158), completed with 22 local objects.
To git@github.com:harrykar/free-programming-books.git
 + 682c429...6e33375 master -> master (forced update)

```
